### PR TITLE
Mark roas lost resources

### PR DESCRIPTION
**Pull request by timbru**

### CODE REVIEW

File: src/commons/bgp/analyser.rs [hunk 1]
```diff
@@ -62,23 +62,44 @@ impl BgpAnalyser {
         }
     }
 
-    pub async fn analyse(&self, roas: &[RoaDefinition], scope: &ResourceSet) -> BgpAnalysisReport {
+    pub async fn analyse(
+        &self,
+        roas: &[RoaDefinition],
+        resources_held: &ResourceSet,
+        limit: Option<ResourceSet>,
```
Comment (ximon18) [inline src/commons/bgp/analyser.rs:69]: Question: Why is `limit` a better name here than `scope`? I find the concept of filtering on a "limit" that "contains" something confusing.

**Comment by ximon18** [comment]
Note: https://github.com/NLnetLabs/lagosta/pull/118 is part of this work.

**Review by ximon18 — APPROVED** [review]
Code here and in the associated Lagosta PR looks good, but I have not properly tested the Lagosta changes.

File: src/commons/bgp/analyser.rs (hunk 1 above)
Comment (ximon18) [reply src/commons/bgp/analyser.rs:69]: Okay. A limit to my mind is a point part way through some spectrum, but in this case it is a set of resources. Would `limited_scope` make sense? I see what you are saying so I suppose this is bikeshedding thus if you're fine with it we can also just leave it as is.

Comment (timbru) [reply src/commons/bgp/analyser.rs]: Sure, if it makes this more clear to you and other readers I am all for it. Pushed.

**Pull request merged**